### PR TITLE
boards: Add Proteus-II board file

### PR DESCRIPTION
**Pull request by mah-eiSmart**
Add a new board file of the Proteus-II radio module

Signed-off-by: Matthias Hauser <Matthias.Hauser@we-online.de>

### CODE REVIEW

**Comment by mah-eiSmart** [comment]
Hey @anangl,

I'm contributing to the Zephyr RTOS for the first time. I've 3 pull requests waiting for review for two of you guys.
Do I have to trigger an action now to get this review? Or do I still wait until you've time for that?

My change to the repository is that I added a board file, that has passed the automatic checks. There should be no risk adding this to the repository.

Thx in advance for explaining.

BR Matthias

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Please squash the fixup changes into the first commit.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please change the commit message to:
```
boards: Add Proteus-II board file

Add a new board file of the Proteus-II radio module

Signed-off-by: Matthias Hauser <Matthias.Hauser@we-online.de>
```

Everything else looks good

**Review by MaureenHelm — DISMISSED** [review]

**Comment by mah-eiSmart** [comment]
Hey together,
the compliance check that failed cannot be solved!

The message of failure is "Boolean option 'BOARD_ENABLE_DCDC' prompt must not start with 'Enable...'. Please
check Kconfig guidelines." but this makro is a makro defined in code of nRFConnectSDK (Nordic Semiconductor). Hence I must use it as they defined it, and cannot change it with respect to the Kconfig guidelines.

Thank you for understanding.

Best regards, Matthias

File: boards/arm/we_proteus2ev_nrf52832/Kconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Proteus-II-EV board configuration
+
+# Copyright (c) 2016 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_WE_PROTEUS2EV_NRF52832
+
+config BOARD_ENABLE_DCDC
+	bool "Enable DCDC mode"
```
Comment (gmarull) [inline boards/arm/we_proteus2ev_nrf52832/Kconfig:9]:
```suggestion
	bool "DCDC mode"
```

**Comment by mah-eiSmart** [comment]
Hello together,

the errors (logging, STM32 ADC driver,...) in the twister-build step are not forced by my pull request. How can I go on to get the pull request merged?

Thx, Matthias

**Comment by gmarull** [comment]
> Hello together,
> 
> the errors (logging, STM32 ADC driver,...) in the twister-build step are not forced by my pull request. How can I go on to get the pull request merged?
> 
> Thx, Matthias

please rebase against upstream main, it may have been fixed. If not please report to #instrastructure.

**Comment by mah-eiSmart** [comment]
> > Hello together,
> > the errors (logging, STM32 ADC driver,...) in the twister-build step are not forced by my pull request. How can I go on to get the pull request merged?
> > Thx, Matthias
> 
> please rebase against upstream main, it may have been fixed. If not please report to #instrastructure.

Hey @gmarull , 
still not fixed, even after rebasing. 

`/__w/zephyr/zephyr/drivers/adc/adc_stm32.c: In function 'adc_stm32_setup_channels':
/__w/zephyr/zephyr/drivers/adc/adc_stm32.c:858:29: error: 'ADC3' undeclared (first use in this function); did you mean 'ADC2'?
  858 |  if ((adc == ADC1 || adc == ADC3) &&
      |                             ^~~~
      |                             ADC2
/__w/zephyr/zephyr/drivers/adc/adc_stm32.c:858:29: note: each undeclared identifier is reported only once for each function it appears in`

What does that mean "report to #instrastructure"? How can I do that?
I removed all dependencies on ADC, but still ADC tests are done.

Thx in advance!
BR Matthias

**Comment by gmarull** [comment]
> > > Hello together,
> > > the errors (logging, STM32 ADC driver,...) in the twister-build step are not forced by my pull request. How can I go on to get the pull request merged?
> > > Thx, Matthias
> > 
> > 
> > please rebase against upstream main, it may have been fixed. If not please report to #instrastructure.
> 
> Hey @gmarull , still not fixed, even after rebasing.
> 
> `/__w/zephyr/zephyr/drivers/adc/adc_stm32.c: In function 'adc_stm32_setup_channels': /__w/zephyr/zephyr/drivers/adc/adc_stm32.c:858:29: error: 'ADC3' undeclared (first use in this function); did you mean 'ADC2'? 858 | if ((adc == ADC1 || adc == ADC3) && | ^~~~ | ADC2 /__w/zephyr/zephyr/drivers/adc/adc_stm32.c:858:29: note: each undeclared identifier is reported only once for each function it appears in`
> 
> What does that mean "report to #instrastructure"? How can I do that? I removed all dependencies on ADC, but still ADC tests are done.
> 
> Thx in advance! BR Matthias

Should have been fixed now, https://github.com/zephyrproject-rtos/zephyr/pull/43752

File: tests/drivers/adc/adc_api/src/test_adc.c [hunk 2]
```diff
@@ -61,6 +61,7 @@
 	defined(CONFIG_BOARD_UBX_EVKNINAB1_NRF52832) || \
 	defined(CONFIG_BOARD_UBX_EVKNINAB3_NRF52840) || \
 	defined(CONFIG_BOARD_UBX_EVKNINAB4_NRF52833) || \
+	defined(CONFIG_BOARD_WE_PROTEUS2EV_NRF52832) || \
```
Comment (gmarull) [inline tests/drivers/adc/adc_api/src/test_adc.c:64]: note: you're not enabling ADC in the board

**Comment by mah-eiSmart** [comment]
Hey, 
I'm still waiting that the bug reported in #43857 is fixed.
As soon this is fixed, I gonna rebase.
BR Matthias

**Review by gmarull — CHANGES_REQUESTED** [review]
All in-tree boards have now been ported to pinctrl, so please do the same here. You can refer to https://github.com/zephyrproject-rtos/zephyr/pull/43926 for more details (includes a script to automate migration).

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**